apps/web/pages/booking/[uid].tsx

```diff
@@ -1068,7 +1073,23 @@ export async function getServerSideProps(context: GetServerSidePropsContext) {
     };
   }
 
-  const bookingInfo = getBookingWithResponses(bookingInfoRaw);
+  const bookingInfoWithSecretResponses = getBookingWithResponses(bookingInfoRaw);
+
+  const filteredResponses = bookingInfoWithSecretResponses.responses;
+
+  if (!session || eventTypeRaw.owner?.id !== session.user.id) {
+    const bookingFields = eventTypeRaw.bookingFields;
+    const parsedBookingFields = bookingFields ? eventTypeBookingFields.parse(bookingFields) : null;
+    parsedBookingFields?.forEach(
+      (field) => field.editable === "user-owner-secret" && delete filteredResponses[field.name]
```

Comment:
Couldn't we check for `field.secretQuestion` here?

Reply:
Instead of doing a delete on `filteredResponses` which is actually a reference of `bookingInfoWithSecretResponses.responses`, we should use array `filter` here.